### PR TITLE
[Toast] New primitive docs

### PR DESCRIPTION
**Pull request by jjenzz**
This still needs a demo but covers all the other areas. It'd be good to see if @peduarte is able to create a demo from following these docs when he is back. 

I've added more detail than we usually do in our accessibility and custom API sections.

### CODE REVIEW

File: components/PropsTable.tsx [hunk 1]
```diff
@@ -78,7 +78,13 @@ export function PropsTable({
...
                     </AccessibleIcon>
                   </PopoverTrigger>
-                  <PopoverContent side="top">
+                  <PopoverContent
+                    side="top"
+                    onOpenAutoFocus={(event) => {
+                      event.preventDefault();
+                      (event.currentTarget as HTMLElement)?.focus();
+                    }}
+                  >
```
Comment (jjenzz) [inline components/PropsTable.tsx:87]: This popover in the `PropsTable` is used to describe the props so may contain links. I've prevented the link from becoming focused and instead focused the content here as it's unlikely these would contain form controls and things.

**Comment by peduarte** [comment]
Nice one. Yeah I can create a demo for this 🚀

File: data/primitives/components/toast/0.1.0.mdx [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+---
+
+# Toast
+
+<Description>A succinct message that is displayed temporarily.</Description>
+
+<HeroContainer>
+  <ToastDemo />
+</HeroContainer>
+
```
Comment (benoitgrelard) [inline data/primitives/components/toast/0.1.0.mdx:15]: I think you're missing the bit that shows the code and the codesanbox link

File: components/HeroContainer.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ export function HeroContainer({ css, children }: { css?: any; children?: React.R
         justifyContent: 'center',
         py: 100,
         borderRadius: '$3',
+        overflow: 'hidden',
```
Comment (andy-hook) [inline components/HeroContainer.tsx:19]: I'd be careful here as not all demos will want to clip this. Perhaps better to handle this inline at the call site?

Comment (jjenzz) [reply components/HeroContainer.tsx:19]: Ah oops! I didn't mean to leave this in. I added it when I had the toast sliding in from the side of the demo but forgot to remove it... good catch 👍

File: components/demos/Toast.jsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+});
+
+// Exports
+const ToastProvider = ToastPrimitive.Provider;
+const ToastViewport = StyledViewport;
+const Toast = StyledToast;
+const ToastTitle = StyledTitle;
+const ToastDescription = StyledDescription;
+const ToastAction = StyledAction;
+const ToastClose = ToastPrimitive.Close;
```
Comment (benoitgrelard) [inline components/demos/Toast.jsx:96]: I think these are usually exported.

Comment (andy-hook) [reply components/demos/Toast.jsx:96]: I was wondering about why that is?

Comment (benoitgrelard) [reply components/demos/Toast.jsx:96]: It's just to simulate what a user might do really, showing that these would be exported from their DS/component file.

Comment (jjenzz) [reply components/demos/Toast.jsx:96]: I'll update the other demos as well then as part of this because I coped this from `Dialog`.

**Pull request merged**